src/connectionconfig/connectionDialogWebViewController.ts

```diff
@@ -106,28 +106,41 @@ export class ConnectionDialogWebViewController extends ReactWebViewPanelControll
 		this.state.connectionProfile = emptyConnection;
 	}
 
-	private async initializeConnectionForDialog(connection: IConnectionInfo) {
+	private async initializeConnectionForDialog(connection: IConnectionInfo): Promise<IConnectionDialogProfile> {
 		// Load the password if it's saved
 		const isConnectionStringConnection = connection.connectionString !== undefined && connection.connectionString !== '';
-		const password = await this._mainController.connectionManager.connectionStore.lookupPassword(connection, isConnectionStringConnection);
 		if (!isConnectionStringConnection) {
+			const password = await this._mainController.connectionManager.connectionStore.lookupPassword(connection, isConnectionStringConnection);
 			connection.password = password;
 		} else {
-			connection.connectionString = '';
-			// extract password from connection string it starts after 'Password=' and ends before ';'
-			const passwordIndex = password.indexOf('Password=') === -1 ? password.indexOf('password=') : password.indexOf('Password=');
-			if (passwordIndex !== -1) {
-				const passwordStart = passwordIndex + 'Password='.length;
-				const passwordEnd = password.indexOf(';', passwordStart);
-				if (passwordEnd !== -1) {
-					connection.password = password.substring(passwordStart, passwordEnd);
+			// If the connection is a connection string connection with SQL Auth:
+			//   * the full connection string is stored as the "password" in the credential store
+			//   * we need to extract the password from the connection string
+			// If the connection is a connection string connection with a different auth type, then there's nothing in the credential store.
+
+			const connectionString = await this._mainController.connectionManager.connectionStore.lookupPassword(connection, isConnectionStringConnection);
+
+			if (connectionString) {
+				const passwordIndex = connectionString.indexOf('Password=') === -1 ? connectionString.indexOf('password=') : connectionString.indexOf('Password=');
```

Comment:
This could be simplified with `const passwordIndex = connectionString.toLowerCase().indexOf('password=');`

The search will only be performed once with this and will match `Password=` regardless of case.